providers/signalk-ws.js

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2016 Ilker Temir <ilker@ilkertemir.com> based on code
+ * by Teppo Kurki <teppo.kurki@iki.fi>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var Transform = require('stream').Transform;
+
+var SignalK = require('signalk-client');
+
+var debug = require('debug')('signalk-server:providers:signalk-ws');
+
+var WebSocket = require('ws');
+var _object = require('lodash/object');
+
+function SignalKWs(options) {
+  Transform.call(this, {
+    objectMode: true
+  });
+  this.selfHost = options.app.config.getExternalHostname() + ".";
+  this.selfPort = options.app.config.getExternalPort();
+  this.remoteServers = {};
+  this.remoteServers[this.selfHost + ":" + this.selfPort] = {};
+  this.signalkClient = new SignalK.Client();
+  var url = "ws://" + options.host + ":" + options.port + options.path;
```

Comment:
I just reviewed the documentation in http://signalk.org/specification/master/connection.html#connection-establishment and I have trouble seeing how it should be clarified. If you have any ideas please let me know.

Only the first http request must be `/signalk`, as the actual REST and ws endpoints can be anything. However I think there is value in sticking to canonical paths to make it easy to connect to any server and find Signal K endpoints in expected places.

If you want to change the discovery process to allow other paths for bootstrapping I think that should be handled as a separate issue, as it is quite beyond adding a static ws provider, the subject of this PR. Bear in mind that to be compatible with that all existing client software needs to be changed.